monkey/monkey_island/setup/setup_param_factory.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+from monkey_island.cc.arg_parser import IslandArgs
+from monkey_island.setup.setup_params import SetupParams
+
+
+class SetupParamFactory:
```

Comment:
I don't think this is an appropriate use of the factory pattern.

> Factory Method is a creational design pattern that provides an interface for creating objects in a superclass, but allows subclasses to alter the type of objects that will be created.
https://refactoring.guru/design-patterns/factory-method

I think it's highly unlikely we'll ever subclass this.

> The factory method pattern is a creational pattern that uses factory methods to deal with the problem of creating objects without having to specify the exact class of the object that will be created.
https://en.wikipedia.org/wiki/Factory_method_pattern

I don't think we'll ever need to take advantage of polymorphism for our config file or command-line parameters. Anything is possible, but this is a bit premature.

It looks like the `SetupParamFactory` class is really just the constructor for `SetupParams`. I'm not sure what the value of externalizing `SetupParams`'s constructor is. It adds some complexity unnecessarily.

If we do want to keep `SetupParams` as a data class, I don't think we need a class that maintains some state in order to construct it. A single function like `getConfig(cmd_args, config_contents)` that calls some other functions is all that's necessary. There's no reason that a `SetupParams` class needs to store `self.setup_params`, as it's never reused after the one and only call to `build()` in `main()`.

Reply:
I agree 100%, this is not a factory pattern. 
The constructor of `SetupParams` is actually there, but hidden. What it does is constructs `SetupParams` with defaults. 
What I attempted to do is separate concerns a bit: if you need to change default parameters, `SetupParamFactory` doesn't need to change; If you need to change the priority of cmd/config file parameters, `SetupParams` doesn't need to change. But it's an overkill for the current situation, probably.

> A single function like getConfig(cmd_args, config_contents) that calls some other functions is all that's necessary.

All other functions use `setup_params`, that's why I refactored it into an object parameter. Otherwise, we need to pass the `setup_params` around. Either way, I'll refactor this back to a single class with SetupParameters